ddtrace/span.py

```diff
@@ -233,16 +233,18 @@ def set_exc_info(self, exc_type, exc_val, exc_tb):
         if not (exc_type and exc_val and exc_tb):
             return # nothing to do
 
-        self.error = 1
+        # readable version of type (e.g. exceptions.ZeroDivisionError)
+        exc_type_str = "%s.%s" % (exc_type.__module__, exc_type.__name__)
+
+        # do not treat celery.exceptions.Retry as an error, as it is expected behavior
+        if exc_type_str != 'celery.exceptions.Retry':
```

Comment:
You can't add celery specific stuff in this file. This should be done at a different level.